haystack/schema.py

```diff
@@ -91,7 +92,13 @@ def __init__(
         self.content_type = content_type
         self.score = score
         self.meta = meta or {}
+        self.id_hash_keys = id_hash_keys
 
+        allowed_hash_key_attributes = ["content", "content_type", "score", "meta", "embedding" ]
+        if not set(self.id_hash_keys) <= set(allowed_hash_key_attributes):
```

Comment:
Hi @ArzelaAscoIi 
I saw mypy having problems with this line because `self.id_hash_keys` could be None. In that case a `TypeError: 'NoneType' object is not iterable` will be raised.
The mypy error is also in the CI tests:
```
haystack/schema.py:98: error: Argument 1 to "set" has incompatible type "Optional[List[str]]"; expected "Iterable[str]"
Found 1 error in 1 file (checked 115 source files)
Error: Process completed with exit code 1.
```
Let's check for id_hash_keys not being None